mlflow/transformers.py

```diff
@@ -1571,7 +1571,50 @@ def _convert_pandas_to_dict(self, data):
                     )
             return parsed
 
-    def predict(self, data, device=None):
+    def _override_inference_config(self, **kwargs):
+        if kwargs:
+            _logger.warning(
+                "kwargs provided to the `predict` method will override the inference configuration "
+                "saved with the model. If the kwargs provided are not valid for the pipeline, "
+                "MlflowException will be raised."
+            )
+
+        # Override the inference configuration with any additional kwargs provided by the user.
+        for key, value in kwargs.items():
+            self.inference_config[key] = value
+
+    def _validate_inference_config_and_return_output(self, data):
+        import transformers
+
+        try:
+            if isinstance(data, dict):
+                return self.pipeline(**data, **self.inference_config)
+            return self.pipeline(data, **self.inference_config)
+        except ValueError as e:
+            if "The following `model_kwargs` are not used by the model" in str(e):
```

Comment:
Relying on an error message is fragile and easier to break. Is this the only way to detect this error?

Reply:
Whatever errors happening when calling self.pipeline(data, **self.inference_config) should be thrown, this error is thrown explicitly by _validate_model_kwargs method in GenerationMixin class https://github.com/huggingface/transformers/blob/4b6a5a7caa1ea31a3321eb17e6dcc9baff4f55d9/src/transformers/generation/utils.py#L1122-L1144

Maybe we need more tests for other supported classes not extending this GenerationMixin class

Reply:
It would have been nice if the maintainers had created a set of custom exceptions for highly specific errors like this, but it seems like pattern matching is the only viable strategy here. 
This exact check and its method content logic has remained unchanged for the lifetime of the 4.x series (as of a refactoring that occurred over 2 years ago), so I think that matching this by message is ok